ZydisEncoder/generate_encoder_types.py

```diff
@@ -0,0 +1,255 @@
+            #define INT32_MIN          (-2147483647 - 1)
+            #define INT64_MIN          (-9223372036854775807 - 1)
+            #define INT8_MAX           127
+            #define INT16_MAX          32767
+            #define INT32_MAX          2147483647
+            #define INT64_MAX          9223372036854775807
+            #define UINT8_MAX          0xff
+            #define UINT16_MAX         0xffff
+            #define UINT32_MAX         0xffffffff
+            #define UINT64_MAX         0xffffffffffffffff""",
+    }
+
+    def __init__(self, zydis_path):
+        super().__init__()
+        self.zydis_path = zydis_path
+        self.add_path(os.path.join(zydis_path, 'dependencies', 'zycore', 'include'))
+        self.add_path(os.path.join(zydis_path, 'include'))
+        self.line_directive = None
+        self.define('__x86_64__')
+        with open(os.path.join(zydis_path, 'src', 'Encoder.c')) as f:
+            self.parse(f.read())
+
+    def on_file_open(self, is_system_include, includepath):
+        if is_system_include:
+            filename = os.path.basename(includepath)
+            if filename in ZydisPreprocessor.FAKE_INCLUDES:
+                return io.StringIO(ZydisPreprocessor.FAKE_INCLUDES[filename])
+            elif filename == 'ZydisExportConfig.h':
+                return open(os.path.join(self.zydis_path, 'assets', 'ZydisExportConfigSample.h'))
+
+        return super().on_file_open(is_system_include, includepath)
+
+    def get_preprocessed_file(self):
+        file = io.StringIO()
+        self.write(file)
+        file.seek(0)
+        return file.read()
+
+    @staticmethod
+    def concatenate_tokens(tokens):
+        return ''.join([token.value for token in tokens])
+
+    def expand_macro(self, macro):
+        if isinstance(macro, str):
+            macro = self.macros[macro].value
+        else:
+            macro = macro.value
+        return self.concatenate_tokens(self.expand_macros(macro))
+
+
+class ZydisParser(object):
+
+    class ZydisParsingError(Exception):
+        pass
+
+    class EnumVisitor(c_ast.NodeVisitor):
+
+        def __init__(self, typenames):
+            self.typenames = typenames
+            self.parsed_types = {}
+
+        @staticmethod
+        def get_const_value(node):
+            if node is None:
+                return None
+            if node.type != 'int':
+                raise ZydisParser.ZydisParsingError('Got enum value of type %s (expected int)' % node.type)
+            if node.value.startswith('0x'):
+                return int(node.value, 16)
+            else:
+                return int(node.value)
+
+        def visit_Typedef(self, node):
+            if node.name in self.typenames:
+                self.parsed_types[node.name] = [(e.name, self.get_const_value(e.value)) for e in node.type.type.values.enumerators
+                                                if not e.name.endswith('_MAX_VALUE') and not e.name.endswith('_REQUIRED_BITS')]
```

Comment:
Tends to simplify your life and aid readability when you break such generator expressions like this:

```suggestion
                self.parsed_types[node.name] = [
                    (e.name, self.get_const_value(e.value)) 
                    for e in node.type.type.values.enumerators
                    if not e.name.endswith('_MAX_VALUE') and not e.name.endswith('_REQUIRED_BITS')
                ]
```